dataactbroker/handlers/fileHandler.py

```diff
@@ -642,3 +727,17 @@ def getProtectedFiles(self):
 
         response["urls"] = self.s3manager.getFileUrls(bucket_name=CONFIG_BROKER["static_files_bucket"], path=CONFIG_BROKER["help_files_path"])
         return JsonResponse.create(StatusCode.OK, response)
+
+    def completeGeneration(self, generationId):
+        """ Retrieve the generated file, using the specified ID to look up submission and file type """
+        if generationId is None:
+            return JsonResponse.error(ResponseException("Must include a generation ID",StatusCode.CLIENT_ERROR), StatusCode.CLIENT_ERROR)
+
+        # Pull url from request
+        safeDictionary = RequestDictionary(self.request)
+        url =  safeDictionary.getValue("href")
+        #Pull information based on task key
+        task = self.interfaces.jobDb.session.query(FileGenerationTask).options(joinedload(FileGenerationTask.file_type)).filter(FileGenerationTask.generation_task_key == generationId).one()
+        job = self.interfaces.jobDb.getJobBySubmissionFileTypeAndJobType(task.submission_id, task.file_type.name, "file_upload")
+        self.load_d_file(url,job.filename,job.original_filename,job.job_id,self.isLocal)
+        return JsonResponse.create(StatusCode.OK,{"message":"File loaded successfully"})
```

Comment:
I'm not sure here, if they re-call the route, should they get an error?  I've already found it useful in debugging to manually re-call these.  In general we've been keeping information for past submissions indefinitely, so I'd be tempted to do the same for these unless we have a reason these should be more temporary.